src/prerequisites/forms.py

```diff
@@ -31,32 +30,20 @@ def popover_labels(model, field_strings):
     return fields_html
 
 
-def hardcoded_prereq_model_choice():
-    """ Can't always dynamically load this list due to accessing contenttypes too early
-    So instead provide a hard coded list which is checked during testing to ensure it matches
-    what the dynamically loaded list would have produced """
-    from courses.models import Block, Course, Grade, Rank
-    from quest_manager.models import Category, Quest
-    from badges.models import Badge
+class AllowedContentObjectChoiceField(ContentObjectChoiceField):
 
-    return [
-        Category, Quest, Block, Course, Grade, Rank, Prereq, Badge,
-    ]
+    def __init__(self, *args, **kwargs):
+        queryset = QuerySetSequence(*[
+            klass.objects.all() for klass in Prereq.all_registered_model_classes()])
+        super().__init__(queryset, *args, **kwargs)
 
 
 class PrereqFormInline(FutureModelForm):
     """This form class is intended to be used in an inline formset"""
 
-    prereq_object = ContentObjectChoiceField(
-        queryset=QuerySetSequence(*[klass.objects.all() for klass in hardcoded_prereq_model_choice()]),
-        widget=CustomContentObjectSelect2Widget(),
-    )
+    prereq_object = AllowedContentObjectChoiceField()
```

Comment:
Similar to above, this will cause errors because content_types framework doesn't exist yet, but the field is being initialized when the code is read, and the __init__ above is trying to access ContentType via `Prereq.all_registered_model_classes()`

Reply:
fixed